src/Models/Communication.php

```diff
@@ -135,4 +173,52 @@ public function getPrintViews(): array
             'communication' => CommunicationView::class,
         ];
     }
+
+    public function autoAssign(string $type, array|string $matchAgainst): void
+    {
+        $matchAgainst = Arr::wrap($matchAgainst);
+        $typeColumn = $type === 'email' ? 'email_primary' : 'phone';
+
+        if ($matchAgainst) {
+            $addresses = resolve_static(Address::class, 'query')
+                ->where(function (Builder $query) use ($matchAgainst, $typeColumn) {
+                    $query->whereIn($typeColumn, $matchAgainst)
+                        ->orWhereHas('contactOptions', function (Builder $query) use ($matchAgainst) {
+                            $query->whereIn('value', $matchAgainst);
+                        });
+                })
+                ->with('contact:id')
+                ->get(['id', 'contact_id'])
+                ->each(
+                    fn (Address $address) => $address->communications()->attach($this->id)
+                );
+
+            $contacts = $addresses->pluck('contact')->unique();
+
+            if ($contacts->isNotEmpty()) {
+                $contacts->each(
+                    fn (Contact $contact) => $contact->communications()->attach($this->id)
+                );
+            }
+        }
+
+        if (is_string($this->subject) && ! empty($this->subject) && $type === 'email') {
```

Comment:
shouldn't this stay in the CreateMailMessage action?

Reply:
nope, because this applies to phone calls as well as mail messages